src/components/PeopleFilters.tsx

```diff
@@ -1,17 +1,137 @@
+import React, { useEffect, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { usePeopleContext } from './PeopleContext/PeopleContext';
+import { calculateBirthCenturies } from './function/calculateBirthCenturies';
+import { SearchLink } from './SearchLink';
+import { filterPeopleByCenturies } from './function/filterPeopleByCenturies';
+import { filterPeopleBySex } from './function/filterPeopleBySex';
+
 export const PeopleFilters = () => {
+  const {
+    searchByName,
+    setSearchByName,
+    setFilteredPeople,
+    people,
+  } = usePeopleContext();
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const century = useMemo(
+    () => searchParams.getAll('centuries') || [], [searchParams],
+  );
+
+  const sexSearch = useMemo(
+    () => searchParams.get('sex'), [searchParams],
+  );
+  const centurysNumber = calculateBirthCenturies(people);
+
+  useEffect(() => {
+    const searchParamValue = searchParams.get('query') || '';
+
+    setSearchByName(searchParamValue);
+
+    const filteredByName = people.filter((person) => (
+      person.name.toLowerCase()
+        .includes(searchParamValue.toLowerCase())
+    ));
+
+    const filteredByCentury = century.length === 0
+      ? filteredByName
+      : filterPeopleByCenturies(filteredByName, century);
+
+    const filteredBySex = !sexSearch
+      ? filteredByCentury
+      : filterPeopleBySex(filteredByCentury, sexSearch);
+
+    setFilteredPeople(filteredBySex);
+  }, [searchParams, people, setSearchByName, setFilteredPeople, century]);
```

Comment:
Try to fix linter error
![image](https://github.com/mate-academy/react_people-table-advanced/assets/56752562/2631c9b5-c659-4850-8b55-a2ad2053117c)

Reply:
Check other linter errors